File: Freedom/我的数据库/MyDatabaseViewController.m
//  MyDatabaseViewController.m
//  Freedom
//  Created by Super on 16/8/18.
//  Copyright © 2016年 Super. All rights reserved.
//
#import "MyDatabaseViewController.h"
#import "MyDatabaseEditViewController.h"
#import "User.h"
@interface DataBaseCollectionViewOCCell : BaseCollectionViewOCCell
@end
@implementation DataBaseCollectionViewOCCell
-(void)initUI{
    self.icon = [[UIImageView alloc]initWithFrame:CGRectMake(10,0, APPW/5-20,40)];
    self.title = [[UILabel alloc]initWithFrame:CGRectMake(0,YH(self.icon), APPW/5-12, 20)];
    self.title.font = fontnomal;self.title.textAlignment = NSTextAlignmentCenter;
    [self addSubviews:self.title,self.icon,nil];
}
-(void)setCollectionDataWithDic:(NSDictionary *)dict{
    self.title.text = [dict valueForKey:@"title"];
    self.icon.image = [UIImage imageNamed:[dict valueForKey:@"icon"]];
}
@end
@interface MyDatabaseViewController()<UICollectionViewDelegateFlowLayout>{
    NSMutableArray *dataArray;
}
@property (nonatomic,strong) UICollectionView *collectionView;
@end
@implementation MyDatabaseViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"我的数据库";
    UIBarButtonItem *left = [[UIBarButtonItem alloc]initWithTitle:@"设置" style:UIBarButtonItemStylePlain actionBlick:^{}];
    UIBarButtonItem *right = [[UIBarButtonItem alloc]initWithImage:[UIImage imageNamed:Padd] style:UIBarButtonItemStyleDone actionBlick:^{}];
    self.navigationItem.leftBarButtonItem  = left;
    self.navigationItem.rightBarButtonItem = right;
    
    UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc]init];
    layout.itemSize = CGSizeMake((APPW-50)/4, 60);
    layout.sectionInset = UIEdgeInsetsMake(0, 10, 0, 10);
    layout.minimumInteritemSpacing = 10;
    layout.minimumLineSpacing = 10;
    
    layout.headerReferenceSize = CGSizeMake(APPW, 30);layout.footerReferenceSize = CGSizeZero;
    self.collectionView = [[UICollectionView alloc]initWithFrame:CGRectMake(0, 0, APPW, APPH-110) collectionViewLayout:layout];
    dataArray = [NSMutableArray arrayWithArray:[User getControllerData]];
    self.collectionView.dataSource = self;
    self.collectionView.delegate = self;
    self.collectionView.frame = self.view.bounds;
    [self.view addSubview:self.collectionView];
}
-(void)viewWillAppear:(BOOL)animated{
    self.tabBarController.tabBar.hidden = YES;
    self.edgesForExtendedLayout = UIRectEdgeLeft | UIRectEdgeRight|UIRectEdgeBottom;
}
- (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath {
    NSString *log = [NSString stringWithFormat:@"你选择的是%zd，%zd", indexPath.section, indexPath.row];
    [SVProgressHUD showSuccessWithStatus:log];DLog(@"%@",log);
    [self pushController:[MyDatabaseEditViewController class] withInfo:dataArray[indexPath.row] withTitle:@"数据库编辑详情" withOther:nil];
}
@end

File: Freedom/我的数据库/MyDatabaseViewController.swift
//
//  MyDatabaseViewController.swift
//  Freedom
import UIKit
import BaseFile
import XExtension
import XCarryOn
class DataBaseCollectionViewCell:BaseCollectionViewCell{
    override func initUI() {
        self.icon = UIImageView(frame: CGRect(x: 10, y: 0, width: APPW/5-20, height:40))
        self.title = UILabel(frame: CGRect(x: 0, y: YH( self.icon), width: APPW/5-12, height:20))
        self.title.font = UIFont.systemFont(ofSize: 14)
        self.title.textAlignment = .center
        addSubviews([self.title,self.icon])
    }
}
class MyDatabaseViewController: MyDatabaseBaseViewController,UICollectionViewDelegate,UICollectionViewDataSource,UICollectionViewDelegateFlowLayout{
    override func viewDidLoad() {
        super.viewDidLoad()
        title = "我的数据库"
        let left = UIBarButtonItem(title: "设置", style: .plain, target: nil, action: nil)
        let right = UIBarButtonItem(image: UIImage(named:"add"), style: .done, target: nil, action: nil)
        self.navigationItem.leftBarButtonItem  = left;
        self.navigationItem.rightBarButtonItem = right;
        let layout = UICollectionViewFlowLayout()
        layout.itemSize = CGSize(width: (APPW-50)/4, height:60)
        layout.sectionInset = UIEdgeInsets(top: 0, left: 10, bottom: 0, right: 10)
        layout.minimumInteritemSpacing = 10;
        layout.minimumLineSpacing = 10;
        layout.headerReferenceSize = CGSize(width: APPW, height:30)
        layout.footerReferenceSize = CGSize.zero
        self.collectionView = BaseCollectionView(frame: CGRect(x: 0, y: 0, width: APPW, height: APPH-110), collectionViewLayout: layout)
//        self.collectionView.dataArray = NSMutableArray(array: nil)
        self.collectionView.dataSource = self;
        self.collectionView.delegate = self;
        self.collectionView.register(DataBaseCollectionViewCell.self, forCellWithReuseIdentifier: DataBaseCollectionViewCell.getColloctionCellIdentifier())
        self.collectionView.backgroundColor = .white
        self.collectionView.frame = self.view.bounds;
        view.addSubview(self.collectionView)
    }
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.tabBarController?.tabBar.isHidden = true
        self.edgesForExtendedLayout = .all
    }
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return 10
    }
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: DataBaseCollectionViewCell.getColloctionCellIdentifier(), for: indexPath) as? DataBaseCollectionViewCell
        cell?.icon.image = UIImage(named:"userLogo")
        cell?.title.text = "name"
        return cell!;
    }
    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
        let log = "你选择的是\(indexPath.section)，\(indexPath.row)"
        noticeInfo(log, autoClear: true, autoClearTime: 3)
        _ = self.push(MyDatabaseEditViewController(), withInfo: "", withTitle: "数据库编辑详情")
    }
}
